server/utils/passport/ensure-authenticated.js

```diff
@@ -1,33 +1,24 @@
-import basePathConfig from '../../configs/basePathConfig'
+import { routes } from '../routes'
 
-const basePath = basePathConfig || ''
-const logoutForbiddenRoute = `${basePath}/logout?e=forbidden`
-const unauthorizedRoute = `${basePath}/logout?e=unauthorized`
-
-export default function ensureAuthenticated(req, res, next) {
-  if (!req.isAuthenticated()) {
-    return res.redirect(`${basePath}/logout`)
+export default async function ensureAuthenticated(req, res, next) {
+  try {
+    if (!req.isAuthenticated()) {
+      return res.redirect(routes.logout)
+    }
+    const { prisma } = req.app.locals
+    const user = await prisma.users.findFirst({
+      where: { uid: req.user },
+      select: { access: true, blocked: true },
+    })
+    switch (true) {
+      case !user || Object.keys(user).length === 0 || !!user.blocked:
```

Comment:
when would we have a user with no attributes (`Object.keys(user).length === 0`)?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/2e3f2e5a20221046bc208d7918023e42684da6e2

Reply:
This commit does not seem relevant to the comment.

Reply:
Sorry about that, next time i'll link to the correct commit.
This should be it
https://github.com/AMP-SCZ/dpdash/pull/241/commits/2e3f2e5a20221046bc208d7918023e42684da6e2#diff-f2aeb466497d5d7bee5a6f7c2154d9f33595759bb92bb4349cce31680a95d2be